### PR TITLE
Update get_ignore_pack_skipped_tests to support tpb from different packs

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/40611

## Description
Updated func `get_ignore_pack_skipped_test`to support ignoring test playbooks that are not related to the pack

### CODE REVIEW

File: demisto_sdk/commands/common/tools.py [hunk 1]
```diff
@@ -830,7 +830,26 @@ def get_pack_ignore_file_path(pack_name):
...
+    for test_playbook_dict in test_playbooks_list:
+        test_playbook_name = list(test_playbook_dict.keys())[0]
+        test_playbook_path = test_playbook_dict[test_playbook_name].get('file_path')
+        test_playbook_pack = test_playbook_dict[test_playbook_name].get('pack')
+
+        if tpb_path in test_playbook_path:
+            return test_playbook_name, test_playbook_pack
+
+
+def get_ignore_pack_skipped_tests(pack_name: str, modified_packs) -> set:
```
Comment (Itay4) [inline demisto_sdk/commands/common/tools.py:852]:
```suggestion
def get_ignore_pack_skipped_tests(pack_name: str, modified_packs: set) -> set:
```

File: demisto_sdk/commands/common/tests/tools_test.py [hunk 2]
```diff
@@ -667,8 +667,9 @@ def test_get_ignore_pack_tests__ignore_test(tmpdir, mocker):
     # prepare mocks
     mocker.patch.object(tools, "get_pack_ignore_file_path", return_value=pack_ignore_path)
     mocker.patch.object(os.path, "join", return_value=str(test_playbook_path / (test_playbook.name + ".yml")))
+    mocker.patch.object(tools, "get_test_playbook_id", return_value=('SamplePlaybookTest', 'FakeTestPack'))
 
-    ignore_test_set = get_ignore_pack_skipped_tests(fake_pack_name)
+    ignore_test_set = get_ignore_pack_skipped_tests(fake_pack_name, {fake_pack_name})
```
Comment (Itay4) [inline demisto_sdk/commands/common/tests/tools_test.py:672]: please update test docstring according to these changes

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -831,7 +831,28 @@ def get_pack_ignore_file_path(pack_name):
...
+
+    Args:
+        test_playbooks_list: The test playbook list from id_set
+        tpb_path: test playbook path.
+
+    Returns (Tuple): test playbook name and pack.
+
+    """
+    for test_playbook_dict in test_playbooks_list:
+        test_playbook_name = list(test_playbook_dict.keys())[0]
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tools.py:845]:
```suggestion
        test_playbook_id = list(test_playbook_dict.keys())[0]
```

File: demisto_sdk/commands/common/tools.py [hunk 4]
```diff
@@ -831,7 +831,28 @@ def get_pack_ignore_file_path(pack_name):
...
+        test_playbook_path = test_playbook_dict[test_playbook_name].get('file_path')
+        test_playbook_pack = test_playbook_dict[test_playbook_name].get('pack')
+
+        if tpb_path in test_playbook_path:
+            return test_playbook_name, test_playbook_pack
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tools.py:850]:
```suggestion
            return test_playbook_id, test_playbook_pack
```

File: demisto_sdk/commands/common/tools.py [hunk 5]
```diff
@@ -864,17 +889,19 @@ def get_ignore_pack_skipped_tests(pack_name: str) -> set:
...
 
+    for item in ignore_list:
+        file_name = item.get('file_name', '')
+        if item.get('ignore_code') == 'auto-test':
+            tests.add(file_name)
+
+            for test in tests:
+                test_id, test_pack = get_test_playbook_id(test_playbooks, test)
+                if test_id:
+                    ignored_tests_set.add(test_id)
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tools.py:904]: why is that inside the previous loop?

**Review by yaakovi — CHANGES_REQUESTED** [review]

**Review by yaakovi — APPROVED** [review]
please fix the LGTM warning

**Pull request merged**